File: dbflow-tests/src/androidTest/java/com/raizlabs/android/dbflow/test/sql/BoxedModel.java
package com.raizlabs.android.dbflow.test.sql;

import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.NotNull;
import com.raizlabs.android.dbflow.annotation.PrimaryKey;
import com.raizlabs.android.dbflow.annotation.Table;
import com.raizlabs.android.dbflow.test.TestDatabase;
import com.raizlabs.android.dbflow.test.structure.TestModel1;

/**
 * Description: Test to ensure that nullable and non-null and boxed primitive classes work as expected.
 */
@Table(database = TestDatabase.class)
public class BoxedModel extends TestModel1 {

    @Column
    @PrimaryKey
    @NotNull
    Long id = 1L;

    @Column
    @NotNull
    public int integerPrimitiveFieldNotNull = 1;

    @Column
    @NotNull
    public Integer integerFieldNotNull = 1;

    @Column
    public Integer integerField = 1;

    @Column
    @NotNull
    public String stringFieldNotNull = "1";

    @Column
    public String stringField = "1";

}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/test/sql/BoxedModel.kt
package com.raizlabs.android.dbflow.test.sql

import com.raizlabs.android.dbflow.annotation.Column
import com.raizlabs.android.dbflow.annotation.NotNull
import com.raizlabs.android.dbflow.annotation.PrimaryKey
import com.raizlabs.android.dbflow.annotation.Table
import com.raizlabs.android.dbflow.test.TestDatabase
import com.raizlabs.android.dbflow.test.structure.TestModel1

/**
 * Description: Test to ensure that nullable and non-null and boxed primitive classes work as expected.
 */
@Table(database = TestDatabase::class)
class BoxedModel : TestModel1() {

    @Column
    @PrimaryKey
    @NotNull
    var id: Long? = 1L

    @Column
    @NotNull
    var integerPrimitiveFieldNotNull = 1

    @Column
    @NotNull
    var integerFieldNotNull: Int? = 1

    @Column
    var integerField: Int? = 1

    @Column
    @NotNull
    var stringFieldNotNull = "1"

    @Column
    var stringField = "1"

}
